deployment/terraform/compute.tf

```diff
@@ -54,96 +42,25 @@ resource "aws_instance" "app_server" {
-      "sudo yum install -y libffi-devel",
-      "curl 'https://bootstrap.pypa.io/get-pip.py' -o '/tmp/get-pip.py'",
-      "sudo python /tmp/get-pip.py",
-      "sudo /usr/local/bin/pip install ansible==2.4.*",
-      "sudo mkdir -p /opt/district-builder/scripts",
-      "sudo mkdir -p /opt/district-builder/deployment/ansible/roles",
-      "sudo chown -R ec2-user:ec2-user /opt/district-builder",
-    ]
-
-    connection {
-      type        = "ssh"
-      host        = "${aws_instance.app_server.private_ip}"
-      user        = "ec2-user"
-      private_key = "${file(pathexpand("~/.ssh/district-builder.pem"))}"
-
-      bastion_host        = "${data.terraform_remote_state.core.bastion_hostname}"
-      bastion_user        = "ec2-user"
-      bastion_private_key = "${file(pathexpand("~/.ssh/district-builder.pem"))}"
-    }
-  }
 }
 
 resource "null_resource" "provision_app_server" {
   triggers {
     uuid = "${uuid()}"
   }
 
-  provisioner "file" {
-    source      = "${path.root}/../user-data"
-    destination = "/opt/district-builder/"
-
-    connection {
-      type        = "ssh"
-      host        = "${aws_instance.app_server.private_ip}"
-      user        = "ec2-user"
-      private_key = "${file(pathexpand("~/.ssh/district-builder.pem"))}"
-
-      bastion_host        = "${data.terraform_remote_state.core.bastion_hostname}"
-      bastion_user        = "ec2-user"
-      bastion_private_key = "${file(pathexpand("~/.ssh/district-builder.pem"))}"
-    }
-  }
-
-  provisioner "file" {
-    source      = "${path.root}/../scripts"
-    destination = "/opt/district-builder"
-
-    connection {
-      type        = "ssh"
-      host        = "${aws_instance.app_server.private_ip}"
-      user        = "ec2-user"
-      private_key = "${file(pathexpand("~/.ssh/district-builder.pem"))}"
-
-      bastion_host        = "${data.terraform_remote_state.core.bastion_hostname}"
-      bastion_user        = "ec2-user"
-      bastion_private_key = "${file(pathexpand("~/.ssh/district-builder.pem"))}"
-    }
-  }
-
-  provisioner "file" {
-    source      = "${path.root}/../ansible/roles/district-builder.app-server"
-    destination = "/opt/district-builder/deployment/ansible/roles"
-
-    connection {
-      type        = "ssh"
-      host        = "${aws_instance.app_server.private_ip}"
-      user        = "ec2-user"
-      private_key = "${file(pathexpand("~/.ssh/district-builder.pem"))}"
-
-      bastion_host        = "${data.terraform_remote_state.core.bastion_hostname}"
-      bastion_user        = "ec2-user"
-      bastion_private_key = "${file(pathexpand("~/.ssh/district-builder.pem"))}"
-    }
-  }
-
-  provisioner "file" {
-    source      = "${path.root}/../ansible/district-builder-app-server.yml"
-    destination = "/opt/district-builder/deployment/ansible/district-builder-app-server.yml"
+  provisioner "remote-exec" {
+    inline = [
+      "while [ ! -f /home/ec2-user/permissions-ready ]; do sleep 2; done",
```

Comment:
I don't think so. The file provisioner runs as soon as SSH is available, so it runs at the same time as cloud-init. Without a wait/signal, we run the risk of trying to move the file into the user-data directory before a) it exists or b) the permissions have been set.

Reply:
But if we just synchronized to a directory like `/home/ec2-user` (and just setup everything there), all of the permissions would be set by default. That would allow us to avoid the need to coordinate for file permission setting, right?

In this scenario, I'm desperately trying to find a way to eliminate the potential for deadlock in the event that some failure prevents the only condition in that infinite loop from being met. I think we could get clever and add a safety valve condition too, but seems best to just get rid of it entirely, if possible.